tests/auth_tests/test_management.py

```diff
@@ -565,3 +565,16 @@ def test_verbose_name_length(self):
         six.assertRaisesRegex(self, exceptions.ValidationError,
             "The verbose_name of auth.permission is longer than 244 characters",
             create_permissions, auth_app_config, verbosity=0)
+
+    def test_custom_permission_name_length(self):
+        auth_app_config = apps.get_app_config('auth')
+
+        ContentType.objects.get_by_natural_key('auth', 'permission')
+        custom_perm_name = 'a' * 256
+        models.Permission._meta.permissions = [
```

Comment:
need to restore the original value of `models.Permission._meta.permissions` in a try/finally block so the test doesn't leak state?